app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -595,21 +599,51 @@ public void prepareFileTransfer() {
             myStatusText.setText(localize("wifi.direct.no.group"));
             return;
         }
+        moveFormRecordsToFiles();
+    }
 
-        zipFiles();
+    private void moveReceivedFiles() {
+        File receiveFolder = new File(toBeSubmittedDirectory);
+        if (!receiveFolder.isDirectory() || !receiveFolder.exists()) {
+            // we haven't received any transfers, just return.
+            return;
+        }
+        // We've received transferred forms. Move them into the to be zipped directory.
+        File[] files = receiveFolder.listFiles();
+        try {
+            Log.d(TAG, "We have " + files.length + " toBeSubmitted files to transfer");
+            for (File instanceDirectory : files) {
+                FileUtil.copyFileDeep(instanceDirectory,
+                        new File(toBeTransferredDirectory, instanceDirectory.getName()));
+                Log.d(TAG, "Transferred " + instanceDirectory + " to " + toBeTransferredDirectory
+                        + "/" + instanceDirectory.getName());
+            }
+        } catch(IOException e){
+            // if we catch an error, delete all the new files we've copied over
+            for (File instanceDirectory : files) {
+                File newDestination = new File(toBeTransferredDirectory, instanceDirectory.getName());
+                if(newDestination.exists()){
+                    FileUtil.deleteFileOrDir(newDestination);
+                }
+            }
+        }
+        // Delete all the to be submitted files, we have them copied for transfer.
+        for (File instanceDirectory : files) {
+            if(instanceDirectory.exists()){
+                FileUtil.deleteFileOrDir(instanceDirectory);
+            }
+        }

```

Comment:
If you follow the advice above, then you can lift the 3 `for` loops in this method into a helper method that essentially looks like this loop, just taking in a `File[] files` param